packages/dashboard-core-plugins/src/panels/MarkdownNotebook.jsx

```diff
@@ -0,0 +1,217 @@
+import React, { PureComponent } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import Markdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+import { Button } from '@deephaven/components';
+import { Code } from '@deephaven/console';
+import { vsPlay } from '@deephaven/icons';
+import './MarkdownNotebook.scss';
+
+export class MarkdownNotebook extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.handleRunSelected = this.handleRunSelected.bind(this);
+    this.renderCodeBlock = this.renderCodeBlock.bind(this);
+    this.renderLink = this.renderLink.bind(this);
+
+    // Map of each code block from it's starting line number to the code within that block
+    this.commands = new Map();
+    this.codeElements = new Map();
+    this.editorScrollView = React.createRef();
+
+    this.state = {
+      hasCode: false,
+
+      // Keep track if any code has been executed yet. If not, make the run button flash
+      hasRunCode: false,
+
+      // Line of the next block to execute. Null to start at the first block
+      nextStartLine: null,
+    };
+  }
+
+  componentDidUpdate() {
```

Comment:
`componentDidUpdate` isn't triggered on the initial render, this only works because the parent updates the props on this component. It doesn't fail so I guess it's ok as is, just a bit flaky.

Reply:
I'll throw it in a componentDidMount as well